scripts/compress.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+import sys
+from ml.metamath.parser import load_database
+from ml.metamath.composer import Composer
+
+
+def number_to_letter(n):
+    number = n - 1
+    final_letter = chr(ord("A") + (number % 20))
+    if number < 20:
+        return final_letter
+
+    number //= 20
+
+    letters = []
+    while True:
+        number -= 1
+        letters.append(chr(ord("U") + ((number % 5))))
+        number //= 5
+        if not number:
+            break
+
+    letters.reverse()
+    letters.append(final_letter)
+    return "".join(letters)
+
+
+def compress(mandatory, proof):
+    label_to_num = {}
+    unique_proof_labels = list(dict.fromkeys(proof))
+    no_mand = [h for h in unique_proof_labels if h not in mandatory]
+    for i, hyp in enumerate(mandatory + no_mand):
+        label_to_num[hyp] = i + 1
+
+    labels_str = " ( " + " ".join(no_mand) + " ) "
+    letters = [number_to_letter(label_to_num[label]) for label in proof]
+    letters_str = "".join(letters)
+    return labels_str + letters_str
+
+
+# assumse that each proof is on a single line
+# Usage: takes in the files from the command line arguments
+# and will output the compressed version to "filename"_compressed
+for file_name in sys.argv[1:]:
```

Comment:
could you use `argparse` here?

Also we could just compress proofs in place for now. Or you can add two arguments `<input> <output>`
which are directories and the script will go through *.mm files in the input directory,
compress, and output files with the same name to the output directory. (this is most convenient way given how our `theory` directory is structued)